packages/near-api-js/tsconfig.json

```diff
@@ -12,7 +12,7 @@
         "alwaysStrict": true,
         "outDir": "./lib",
         "declaration": true,
-        "preserveSymlinks": true,
+        "preserveSymlinks": false,
```

Comment:
I have encountered a weird bug where adding `ajv` dependency messes up with `pnpm`'s dependency model. It can't find some of the `ajv` transitive dependencies even though they are clearly there. I and @morgsmccauley have debugged the issue and found out that toggling this line fixes this.

Reply:
Adding more context while it's fresh:

This seems to be an issue within `pnpm` that has already been fixed: https://github.com/pnpm/pnpm/issues/496, so not sure why this is still a problem.

`pnpm` will place all 'visible' dependencies in their own `node_modules` folder and then symlink to that directory from the top level:
```
node_modules/
|  ajv ➛ node_modules/.pnpm/ajv@8.11.2/node_modules/ajv
|  .pnpm/
|  |  ajv@8.11.2/
│  |  |  node_modules/
|  │  |  |  ajv/
|  |  |  |  fast-deep-equal/ ➛ node_modules/.pnpm/fast-deep-equal@3.1.3/node_modules/fast-deep-equal
|  |  |  |  json-schema-traverse/ ➛ node_modules/.pnpm/json-schema-traverse@1.0.0/node_modules/json-schema-traverse
|  |  |  |  require-from-string/ ➛ node_modules/.pnpm/require-from-string@2.0.2/node_modules/require-from-string
|  |  |  |  uri-js/ ➛ node_modules/.pnpm/uri-js@4.4.1/node_modules/uri-js
```
This strategy makes all transitive dependencies inaccessible from your package code while still making them accessible to the dependency itself. In the above example, all of `ajv`s dependencies are next to it (in the bottom most directory), meaning `ajv` should be able to `import`/`require` it.

But when we use TS to build, we get the following error, which complains that `ajv` can't access `uri-js`. This doesn't make any sense because we can see it within `node_modules/`:
```sh
> near-api-js@1.1.0 compile /Users/morganmccauley/Developer/Repositories/near-api-js/packages/near-api-js
> tsc -p ./tsconfig.json

node_modules/ajv/dist/compile/resolve.d.ts(3,36): error TS2307: Cannot find module 'uri-js' or its corresponding type declarations.
node_modules/ajv/dist/types/index.d.ts(1,22): error TS2307: Cannot find module 'uri-js' or its corresponding type declarations.
node_modules/near-abi/lib/index.d.ts(1,29): error TS2307: Cannot find module 'json-schema' or its corresponding type declarations.
```

Setting [preserveSymlinks](https://www.typescriptlang.org/tsconfig#preserveSymlinks) to `false` is counter-intuitive here, but works.. When disabled, module resolution happens from where the symbolic link is defined, rather from where it actually resolves to. I would think that we would want the latter, since that is where the transitive dependencies are defined, but this doesn't seem to be the case.

maybe we should raise this issue within `pnpm` itself 🤔